dist/api_data.py

```diff
@@ -1162,6 +1162,14 @@ def __cmp__(self, other):
         type='list'),
 ]),
 
+'WT_SESSION.query_timestamp' : Method([
+    Config('get', 'read_timestamp', r'''
+        specify which timestamp to query: \c read_timestamp returns the
+        timestamp the transaction is reading at.  See @ref
+        transaction_timestamps''',
+        choices=['read_timestamp']),
```

Comment:
I have two comments on the API:
1. I have a preference for it to just be `read` as the `timestamp` part is implied in the method name, and it is consistent with the usage of the connection's `query_timestamp` of `oldest, pinned` etc.
2. Now with all the infrastructure in place, I'd like to see support for all the timestamps available in the txn. May as well add them now and be complete. `commit|first_commit|prepare|read`